lib/elixir/lib/kernel.ex

```diff
@@ -2703,7 +2703,15 @@ defmodule Kernel do
   """
   defmacro left |> right do
     [{h, _}|t] = Macro.unpipe({:|>, [], [left, right]})
-    :lists.foldl fn {x, pos}, acc -> Macro.pipe(acc, x, pos) end, h, t
+    :lists.foldl fn {x, pos}, acc ->
+      # TODO: raise an error in `Macro.pipe` when we drop unary operator support in pipes
+      case Macro.pipe_warning(x) do
```

Comment:
What if we make `Macro.pipe_warning/1` accept a second argument, the caller? So that it can take care of actually warning as well:

```elixir
def pipe_warning({call, _, _}, caller) when call in unquote(@unary_ops) do
  :elixir_errors.warn(caller.line, caller.file, "piping into a unary operator is deprecated")
end
```